binderhub/repoproviders.py

```diff
@@ -69,6 +69,12 @@ class RepoProvider(LoggingConfigurable):
 
     unresolved_ref = Unicode()
 
+    git_credentials = Unicode(
+        "",
+        help="""
+        Credentials (if any) to pass to git when cloning.
```

Comment:
Do we mean a token for GitHub or GitLab?

Reply:
Nope, these are directly used by git, not specific to GitHub / GitLab. In the repoproviders, we provide the particular format GitHub requires for its git credentials, and when we support private GitLab it would need a similar formatter.